art/attacks/evasion/simba.py

```diff
@@ -107,9 +108,15 @@ def generate(self, x: np.ndarray, y: Optional[np.ndarray] = None, **kwargs) -> n
         x = x.astype(ART_NUMPY_DTYPE)
         x_adv = x.copy()
 
-        preds = self.estimator.predict(x, batch_size=self.batch_size)
+        y_prob_pred = self.estimator.predict(x, batch_size=self.batch_size)
 
-        if self.estimator.nb_classes == 2 and preds.shape[1] == 1:
+        if not is_probability(y_prob_pred):
+            raise ValueError(
+                "This attack requires an estimator predicting probabilities. It looks like the current "
```

Comment:
This is definitely helpful but maybe change the error message to recommend a fix?
```
                "This attack requires an estimator predicting probabilities. Try adding 
                a Softmax layer to your model and/or using the `art.BlackBoxEstimator` class"
```

Reply:
@Embeddave Yes, that should be helpful. I'm not sure if I understand `art.BlackBoxEstimator`?

Reply:
I'm not sure if I understand either 🙂 

I should have said `BlackBoxClassifier`

In truth it's not clear to me whether that would be appropriate and/or expected to use with this attack -- I thought I had read an example notebook that used the `BlackBoxClassifier` wrapper with a similar attack but now I can't find it

So, nevermind :)

but a concrete solution like "add softmax layer" I think we agree could be good?